src/Platform/Microsoft.Testing.Platform/DotnetTest/DotnetTestConnection.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Testing.Platform.CommandLine;
+using Microsoft.Testing.Platform.Extensions.CommandLine;
+using Microsoft.Testing.Platform.IPC;
+using Microsoft.Testing.Platform.IPC.Models;
+using Microsoft.Testing.Platform.IPC.Serializers;
+using Microsoft.Testing.Platform.Services;
+using Microsoft.Testing.Platform.Tools;
+
+namespace Microsoft.Testing.Platform.DotnetTest;
+
+internal sealed class DotnetTestConnection
+{
+    private readonly CommandLineHandler _commandLineHandler;
+    private readonly ITestApplicationModuleInfo _testApplicationModuleInfo;
+    private readonly CTRLPlusCCancellationTokenSource _cancellationTokenSource;
+
+    public DotnetTestConnection(CommandLineHandler commandLineHandler, ITestApplicationModuleInfo testApplicationModuleInfo, CTRLPlusCCancellationTokenSource cancellationTokenSource)
```

Comment:
I would suggest to use the interface instead of the implementation class for the token source.